lib/core/test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -857,3 +859,14 @@ postTransactionViaCLI ctx passphrase args = do
 proc' :: FilePath -> [String] -> CreateProcess
 proc' cmd args = (proc "stack" (["exec", "--", cmd] ++ args))
     { std_in = CreatePipe, std_out = CreatePipe, std_err = CreatePipe }
+
+-- | Get process out and err streams after n seconds of running and terminate
+getProcStream :: CreateProcess -> Int -> IO (String, String)
+getProcStream pr n = withCreateProcess pr $ \_ (Just stdout) (Just stderr) h -> do
+       threadDelay $ n*oneSecond
```

Comment:
We should avoid delays as much as possible for these kind of tests because it creates very flaky tests in the end. It depends on many things in the host environment, and we often are too short in the delay. Instead, I'd suggest to follow a similar strategy to what we did for the file watching: 

- Wait for some event to appear
- Timeout the whole waiting time with a rather big upper-bound, like a minute or so.

So here, we could wait until we've seen in the log stream some lines, and wait at most 60s. If after 60s we haven't seen anything, then the test has failed.

Reply:
>So here, we could wait until we've seen in the log stream some lines, and wait at most 60s. If after 60s we haven't seen anything, then the test has failed.

Gotcha, however this wouldn't work for `--quiet` as it does not log anything in normal circumstances (ofc except for errors)

Reply:
Ah! Well, for the `--quiet`, maybe having a slightly bigger timeout (like 10s) to really leave time for some logs to appear?

Reply:
Hm, well `--quiet` logs only `error` so nothing is basically logged when everything is fine... perhaps one way is producing some error artificially to have something in the log (¯\_(ツ)_/¯) or just leave it as is for this particular case...

Reply:
What I mean is that, if it doesn't log anything after one seconde, we can't be sure that it has even started. If it doesn't log anything after 10, we still aren't sure, but at least it increases a bit our confidence.